In [1]:
import pandas as pd
import numpy as np
from scipy import stats

# Problem Statement 1:
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions

Is there significant difference in the diameter of the cutlet ?

Data description
α == 0.05 (95% Confidence)

Y == Continuous
X == Discrete

Is Y1 and Y2 normal ?

H0 = Y1 and Y2 are normal
H1 = Y1 and Y2 are not normal

# Normality Test

In [2]:
df = pd.read_csv("C:/Users/DIKSHA/Downloads/Cutlets.csv")

In [3]:
stats.shapiro(df["Unit A"])

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)

P value for Unit A == 0.32 > α

In [4]:
stats.shapiro(df["Unit B"])

ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)

P value for Unit B == 0.52 > α

HO is accepted. Thats is both Y1 and Y2 are normal

External condition are same.
Thus we can perform Paired T Test

H0 == Mean for Y1 and Y2 are equal (There is no significance difference between diameter of the Culets)
H1 == Mean for Y1 and Y2 are not equal (There is a significance difference between diameter of the Culets)

In [5]:
stats.ttest_rel(df["Unit A"], df["Unit B"])

Ttest_relResult(statistic=0.7536787225614323, pvalue=0.45623007680384076)

P value of the Paired T Test is == 0.45 > α

# Thus H0 is accepted.
Mean of both Y1 and Y2 are equal

# Problem Statement 2:
A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level

α == 0.05 (95% Confidence)

Y == Continuous
X == Discrete

Is Y1, Y2, Y3 and Y4 normal ?

H0 = Y1, Y2, Y3 and Y4 are normal
H1 = Y1, Y2, Y3 and Y4 are not normal

# Normality Test

In [6]:
df = pd.read_csv("C:/Users/DIKSHA/Downloads/LabTAT.csv")

In [7]:
stats.shapiro(df["Laboratory 1"])

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)

P value for Laboratory 1 == 0.55 > α

In [8]:
stats.shapiro(df["Laboratory 2"])

ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)

P value for Laboratory 2 == 0.86 > α

In [10]:
stats.shapiro(df["Laboratory 3"])

ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)

P value for Laboratory 3 == 0.42 > α

In [11]:
stats.shapiro(df["Laboratory 4"])

ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)

P value for Laboratory 4 == 0.66 > α

HO is accepted. Thats is Y1, Y2, Y3, and Y4 are normal

# VARIANCE

H0 == Variance of all 4 laboratories are the same
H1 == Variance of all 4 laboratories are the not same

In [12]:
stats.levene(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

P Value of Variance test is == 0.051 > α

H0 is accepted
Thus we will perform one way Anova Test

In [13]:
stats.stats.f_oneway(df["Laboratory 1"], df["Laboratory 2"], df["Laboratory 3"], df["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

P value of the One way Anova test is == 2.1156< α

# Thus H1 is accepted.
Mean TAT for 4 laboratories not equal (There is a significance difference between TAT of the laboratories)

# Problem Statement 3:
Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions

Is the male-female buyer rations are similar across regions

α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test

# Chi-Square Test

In [14]:
df = pd.read_csv("C:/Users/DIKSHA/Downloads/BuyerRatio.csv")

H0 == The male-female buyer rations are similar across regions
H1 == The male-female buyer rations are not similar across regions

In [15]:
stats.chi2_contingency([df["East"], df["West"], df["North"], df["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

P value of Chi-Square test == 0.66 > α

# Thus HO is accepted.
The male-female buyer rations are similar across regions

# Problem Statement 4:
TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Does the defective % varies significantly by centre ?

α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test


# Chi-Square Test

In [16]:
df = pd.read_csv("C:/Users/DIKSHA/Downloads/Costomer+OrderForm.csv")

H0 == The defective % does not varies by centre
H1 == The defective % does varies by centre

In [17]:
stats.chi2_contingency([df['Phillippines'].value_counts(), df['Indonesia'].value_counts(), df['Malta'].value_counts(), df['India'].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

P value of Chi-Square test == 0.277 > α

# Thus HO is accepted.
The defective % does not varies significantly by centres